### ISSUE TITLE
Weird Problem when bundeling specific library (js-base64)

### ISSUE THREAD

**Issue by Belkar** [systemjs/builder#659]
The problem occured with the library **angular2-oauth2** and its dependency js-base64.

As long as I don't bundle this library up everything works. But as soon as I try to add it to my bundle, SystemJS isn't able to interprete `base64.js` (from js-base64) as a module.

I prepared a small test project with solely a call to js-base64 that demonstrates the problem. The dependency to angular2-oauth2 is in there, but only for the dependencies and not the library itself.

[bundleIssue.zip](https://github.com/systemjs/builder/files/395557/bundleIssue.zip)

The Project is strapped from a real project with only the parts that affect the issue. It consists of an `index.html` file that loads SystemJS and the configuration as well as the bundled dependencies and starts the application.
The application itselfs only display a decoded base64 string.
Additionally there are some gulp-tasks to compile the TypeScript files, copy the dependencies to the `wwwroot/lib`-Folder and start the bundeling script.

---

**tl;dr**: The Bundled library does not behave like the unbundled version

**Comment by guybedford** [comment]
I hope you can understand I don't have the time to deeply look into your replication case here.

What is the actual error you are getting here? Is the error during the build or when running the production environment?

**Comment by Belkar** [comment]
When running the web application in the browser, i get the following error message:
```
Error: (SystemJS) Cannot read property 'decode' of undefined
	TypeError: Cannot read property 'decode' of undefined
	    at Object.eval (http://127.0.0.1:8080/app/boot.js:3:25)
	    at eval (http://127.0.0.1:8080/app/boot.js:7:4)
	    at eval (http://127.0.0.1:8080/app/boot.js:8:3)
	Evaluating http://127.0.0.1:8080/app/boot.js
	Error loading http://127.0.0.1:8080/app/boot.js
```

When debugging the code, you can see that the object, which is returned by `require("js-base64")`, is empty.

If I run the application without bundling the dependencies, the call of `require` returns the desired object.

**Comment by guybedford** [comment]
@Belkar try using `import Base64 from "js-base64"` here. You'd have to use `import {fromByteArray} from 'js-base64'` anyway for the import to make sense surely?

**Comment by Belkar** [comment]
`require('js-base64')` is generated from `import {Base64} from "js-base64";` using TypeScript and `fromByteArray` is not needed.

Another problem is, that this call is inside the third party library `angular2-oauth2` (and there fore I can't really modify it like you suggested). I wanted to try if it also happens when called outside the library so I created this little base program (so there is no angular context around it, as it may interfere).

**Comment by guybedford** [comment]
Ensure you're using the `system` format for TypeScript compilation.

**Comment by guybedford** [comment]
In CommonJS, `require('js-base64')` is the object `{ fromByteArray: ..., ... }`.

**Comment by Belkar** [comment]
Also no luck. :disappointed:

**Comment by TianCNCA** [comment]
I'm having the exact same problem.

**Comment by Belkar** [comment]
Not really. We "fixed" it by excluding js-base64 with bundeling arithmetic like so:
`builder.bundle("app/**/* - [js-base64]", "deps.js", { minify: false, sourceMaps: true });`

We are now planning on migrating to Webpack since it looks more promising and angular also has growing tendendies to use it.
